api/app/jobs/noaa.py

```diff
@@ -117,12 +191,51 @@ def parse_url_for_timestamps(url: str):
     return (model_run_timestamp, prediction_timestamp)
 
 
+def parse_nam_url_for_timestamps(url: str):
+    """ Interpret the model_run_timestamp and prediction_timestamp from a NAM model's URL """
+    # sample URL: 'https://nomads.ncep.noaa.gov/cgi-bin/filter_nam_na.pl?dir=%2Fnam.20230414&file=nam.t00z.awip3220.tm00.grib2&var_APCP=on&var_RH=on&var_TMP=on&var_UGRD=on&var_VGRD=on&lev_surface=on&lev_2_m_above_ground=on&lev_10_m_above_ground=on&subregion=&toplat=60&leftlon=-139&rightlon=-114&bottomlat=48'
+    model_run_date_str = re.search("dir=\%2Fnam.\d*", url).group(0)
```

Comment:
Can we just use `urllib`'s `parse_qs` to extract these values?

https://stackoverflow.com/a/43628262

Reply:
Refactored this method and the equivalent one for GFS to use `parse_qs`